pkgs/applications/networking/cluster/pig/default.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv, fetchurl, makeWrapper, hadoop, jre, bash }:
+
+stdenv.mkDerivation rec {
+
+  name = "pig-0.14.0";
+
+  src = fetchurl {
+    url = "mirror://apache/pig/${name}/${name}.tar.gz";
+    sha256 = "183in34cj93ny3lhqyq76g9pjqgw1qlwakk5v6x847vrlkfndska";
+
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  buildPhase = ''
+    for n in "bin/"* "sbin/"*; do
+      sed -i $n -e "s|#!/usr/bin/env bash|#! ${bash}/bin/bash|"
+    done
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    mv * $out
+
+    # no need for the windows batch script
+    rm $out/bin/pig.cmd $out/bin/pig.py
+
+    for n in $out/{bin,sbin}"/"*; do
+      wrapProgram $n --prefix PATH : "${jre}/bin:${bash}/bin" --set JAVA_HOME "${jre}" --set HADOOP_PREFIX "${hadoop}"
```

Comment:
(style) you could tidy this with backslash and have --prefix and --set each on its own line